bookmarks/models.py

```diff
@@ -7,11 +7,31 @@
 
 
 class Bookmark (models.Model):
-    user = models.ForeignKey(User, on_delete=models.CASCADE)
-    course = models.ForeignKey(Course, on_delete=models.CASCADE, default='SOME STRING')
+    user = models.ForeignKey(User, on_delete=models.PROTECT)
+    course = models.ForeignKey(Course, on_delete=models.PROTECT)
     url = models.URLField(max_length=200)
     urlname = models.CharField(max_length=30)
-    last_view_date = models.DateTimeField(auto_now=True, null=True)
+    last_view_date = models.DateTimeField(auto_now=True, blank=True)
 
     def __str__(self):
         return self.urlname
+
+    @classmethod
+    def add_bookmark(cls, user, course, url, urlname):
+        """Creates new bookmark"""
+        b = Bookmark.objects.create(user=user, course=course, url=url, urlname=urlname)
+        b.save()
+        pass
```

Comment:
1) when using objects.create, I don't think you need to call save.
2) When adding a new bookmark, don't you want to return the newly created object? (return b)

Reply:
1. .save() perform internally as either INSERT or UPDATE object to db, while .objects.create() perform only INSERT object to db.
2. No, I just wanted to make sure the new object is saved to db propely, while in the tests I pull the existing data using a filter.

Reply:
1. Fixed it throughout the code (It's was appearing that way in the test function as well) 
2. In this function I just wanted to make sure that the object is saved to the db while in the test i'm pulling the object from the db using a filter.
@simzacks

Reply:
ok. In general, think about actual usage and whether most of the time the user will actually want the object after creating it or not. If so, you'll want a return instead of a save and then a filter.